cardano-tx-generator/test/Main.hs

```diff
@@ -1,106 +0,0 @@
-  [ testCase "tx-send == tx-received" $ assertBool "tx-send == tx-received" True -- TODO !
-  ]
-
-sizedMetadata = testGroup "properties of the CBOR encoding relevant for generating sized metadat"
-  [ testCase "Shelley metadata map costs" $ assertBool "metadata map costs" prop_mapCostsShelley
-  , testCase "Shelley metadata ByteString costs" $ assertBool "metadata ByteString costs" prop_bsCostsShelley
-  , testCase "Allegra metadata map costs" $ assertBool "metadata map costs" prop_mapCostsAllegra
-  , testCase "Allegra metadata ByteString costs" $ assertBool "metadata ByteString costs" prop_bsCostsAllegra
-  , testCase "Mary metadata map costs"    $ assertBool "metadata map costs" prop_mapCostsMary
-  , testCase "Marymetadata ByteString costs"    $ assertBool "metadata ByteString costs" prop_bsCostsMary
-  , testCase "Test mkMetadata" $ assertBool "" True --WIP
-  ]
-
-
-cliArgs = testGroup "cli arguments"
-  [
-     -- Also update readme and documentation when the help-messages changes.
-    testCase "check help message against pinned version"
-      $ assertBool "help message == pinned help message" $ helpMessage == pinnedHelpMessage
-
-     -- examples for calling the tx-generator found in the shell scripts.
-  , testCmdLine [here|--config /work/cli-tests/benchmarks/shelley3pools/configuration/configuration-generator.yaml --socket-path /work/cli-tests/benchmarks/shelley3pools/logs/sockets/1 --num-of-txs 1000 --add-tx-size 0 --inputs-per-tx 1 --outputs-per-tx 1 --tx-fee 1000000 --tps 10 --init-cooldown 5 --target-node ("127.0.0.1",3000) --target-node ("127.0.0.1",3001) --target-node ("127.0.0.1",3002) --genesis-funds-key configuration/genesis-shelley/utxo-keys/utxo1.skey|]
-  ]
-  where
-    testCmdLine :: String -> TestTree
-    testCmdLine l = testCase "check that example cmd line parses" $ assertBool l $ isJust
-                        $ getParseResult $ execParserPure defaultPrefs (info parseCommand fullDesc)
-                           $ words l
-
-pinnedHelpMessage = [here|ParserFailure(Usage: <program> --config FILEPATH --socket-path FILEPATH 
-                 [--shelley | --mary | --allegra] [(--target-node (HOST,PORT))] 
-                 [--init-cooldown INT] [--initial-ttl INT] [--num-of-txs INT] 
-                 [--tps DOUBLE] [--inputs-per-tx INT] [--outputs-per-tx INT] 
-                 [--tx-fee INT] [--add-tx-size INT] 
-                 [--fail-on-submission-errors] [--n2n-magic-override NATURAL] 
-                 [--addr-mainnet] 
-                 (--genesis-funds-key FILEPATH | --utxo-funds-key FILEPATH
-                   --tx-in TX-IN --tx-out TX-OUT |
-                   --split-utxo-funds-key FILEPATH --split-utxo FILEPATH)
-
-Available options:
-  --config FILEPATH        Configuration file for the cardano-node
-  --socket-path FILEPATH   Path to a cardano-node socket
-  --shelley                Initialise Cardano in shelley submode.
-  --mary                   Initialise Cardano in mary submode.
```

Comment:
This was just added for multi era support.
Git blame on the testcase show exactly when CLI arguments change.
IHMO that is useful.

Reply:
I'm of a split mind here -- this really increases the cost of change.

And in a recent discussion we've agreed that it's _already_ too painful to change the CLI, if you remember.